MigrationV3V4/CSharp/Skills/V3SimpleSandwichBot/Controllers/MessagesController.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using System.Web.Http;
+
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Builder.FormFlow;
+using System.Net.Http;
+using System.Web.Http.Description;
+using System.Diagnostics;
+using Microsoft.Bot.Connector.SkillAuthentication;
+
+namespace Microsoft.Bot.Sample.SimpleSandwichBot.Controllers
+{
+    [SkillBotAuthentication]
+    public class MessagesController : ApiController
+    {
+        internal static IDialog<SandwichOrder> MakeRootDialog()
+        {
+            return Chain.From(() => FormDialog.FromForm(SandwichOrder.BuildForm));
+        }
+
+        /// <summary>
+        /// POST: api/Messages
+        /// receive a message from a user and send replies
+        /// </summary>
+        /// <param name="activity"></param>
+        [ResponseType(typeof(void))]
+        public virtual async Task<HttpResponseMessage> Post([FromBody] Activity activity)
+        {
+            if (activity != null)
+            {
+                // one of these will have an interface and process it
+                switch (activity.GetActivityType())
+                {
+                    case ActivityTypes.Message:
+                        if (activity.Text.ToLower().Contains("end") || activity.Text.ToLower().Contains("stop"))
+                        {
+                            await SkillsHelper.EndSkillConversation(activity);
+                        }
+                        else
+                        {
+                            await Conversation.SendAsync(activity, MakeRootDialog);
+                        }
+
+                        break;
+
+                    case ActivityTypes.ConversationUpdate:
+                    case ActivityTypes.ContactRelationUpdate:
+                    case ActivityTypes.Typing:
+                    case ActivityTypes.DeleteUserData:
+                    default:
```

Comment:
We should handle an incoming `endOfConversation` activity, as the skill consumer can send this to cancel the current conversation. The skill should "release resources" as appropriate (clear state, etc.).

Reply:
Thanks, added code for endOfConversation from the skill host